### PR TITLE
Set `link_power_policy_management` default to "max_performance"

### PR DESCRIPTION
**Pull request by aaronlevin**
The kernel default for `link_power_management_policy` is `"max_performance"`. This commit: https://github.com/NixOS/nixpkgs/commit/f169f60575bec7c61626af180cae364d321b4bec set the NixOS default to `"min_power"`.

This issue (https://github.com/NixOS/nixpkgs/issues/11276) details my long journey to discover that, after several file system failures incorrectly attributed to `TRIM` and `NCQ` settings, the origin of my (and others') issues were around the `"min_power"` default setting. Setting it back to `"max_performance"` as per the kernel default resolved my issues.

I think we should use the kernel default of `"max_performance"` to assure the best experience for new users with SSDs and to conform to the defaults of the kernel and other distros.

### CODE REVIEW

**Comment by mbakke** [comment]
IMO the `powerManagement` interface stinks and should not be enabled by default, particularly on Sandy Bridge and newer. At least it should not unconditionally include cpufreq as it directly conflicts with the pstate driver.

SCSI link power policy is a small improvement on a laptop, but useless on a desktop. I'm running `powertop --auto-tune` as a boot service (on laptops) and have `powerManagement.enable = false` (on all systems).

Sorry for the semi-related rant. I'm starting to wonder whether cpufreq may actually have contributed to the problems we saw in the other issue.

**Comment by domenkozar** [comment]
cc @chaoflow

**Comment by abbradar** [comment]
I think that the solution might be to not set both `cpuFreqGovernor` and `scsiLinkPolicy` by default at all. We can't deduce proper governor anyway (for me, for instance, as a Haswell user) and `scsiLinkPolicy` can cause problems, too. Other things in `powerManagement` (and these too if set explicitly) are useful and enabled-by-default behavior is good IMHO, if not for these defaults.

On the side note, we also have `tlp` service for dynamically setting power management options on ACPI events.

**Comment by edolstra** [comment]
Well, if setting the link power management policy to `min_power` results in filesystem corruption, it's a kernel bug and should be fixed in the kernel. Was this issue reported upstream?

File: nixos/modules/config/power-management.nix [hunk 1]
```diff
@@ -71,7 +71,7 @@ in
 
     # FIXME: Implement powersave governor for sandy bridge or later Intel CPUs
     powerManagement.cpuFreqGovernor = mkDefault "ondemand";
-    powerManagement.scsiLinkPolicy = mkDefault "min_power";
+    powerManagement.scsiLinkPolicy = mkDefault "max_performance";
```
Comment (edolstra) [inline nixos/modules/config/power-management.nix]: Why not just remove this line altogether, given that this is the kernel default?

Comment (aaronlevin) [inline nixos/modules/config/power-management.nix]: I can do that. I wasn't quite sure how these defaults work. will update when I get home.

**Comment by aaronlevin** [comment]
@edolstra I wasn't able to find a bug report that wasn't a few years old (e.g.: https://bugzilla.kernel.org/show_bug.cgi?id=14543). Perhaps the `"max_performance"` default came out of one of these issues? Not clear to me.

I don't want to project my own difficulties onto the project, but I almost gave up running NixOS because of these failures. Even if it's an upstream bug, our current default, different than the kernel, is causing people problems.

**Comment by nckx** [comment]
So I just discovered this service after noticing my kernel's default CPU frequency governor was being happily ignored...

> @mbakke: IMO the powerManagement interface stinks

:clap: 

> and should not be enabled by default, particularly on Sandy Bridge and newer. At least it should not unconditionally include cpufreq as it directly conflicts with the pstate driver.

It shouldn't run at all. Not by default.

Whether it should exist at all is in its current form is debatable.

Whether NixOS, by default, silently overrides _both kernel defaults and explicit configuration_ really isn't – orthogonal kernel bugs aside.

**Pull request merged**